### ISSUE TITLE
Unexpected white space before footer

### ISSUE THREAD

**Issue by djtsorrell** [koreader/koreader#7769]
* KOReader version: 2021.05
* Device: Kobo Libra H2O

I have noticed that the white space between the last line and the footer is inconsistent. Top and bottom margins are both set to 15 pt and the status bar container height is set to 20 pt. The `Reclaim bar height from bottom margin` setting is enabled. Disabling that setting still results in inconsistent gaps. I have tried editing line spacing and font size to no avail. Sometimes, the text clips the status bar. 
Here are some screenshots to demonstrate the problem.

## Excess white space at the end of page
![Reader_2021-05-27](https://user-images.githubusercontent.com/45468834/119896697-4d8b1880-bf37-11eb-800a-12c04c510895.png)

## No excess white space at the end of the next page
![Reader_2021-05-27_214113](https://user-images.githubusercontent.com/45468834/119895215-6e526e80-bf35-11eb-8e68-9be465cad9a0.png)

I believe there is enough space at the end of the first page for another line.

**Comment by NiLuJe** [comment]
> The `Reclaim bar height from bottom margin` setting is enabled.
> [...]
> Sometimes, the text clips the status bar.

That's entirely normal with that setting enabled, in fact, it's the *expected* behavior ;).

> I believe there is enough space at the end of the first page for another line.

You haven't shown us your margin settings, but assuming they're balanced, given your top margin and lien-height, no, there isn't ;).

**Comment by NiLuJe** [comment]
Might be down to a couple of pixels, but, even with very generous approximations:

![meeep](https://user-images.githubusercontent.com/111974/119899700-c3957c80-bf43-11eb-8e11-98607bccb3da.png)

* Blue is the line gap
* Green is the margin
* Red is (an approximation of) a glyph's bounding box.

Not enough height to fit red while still honoring green: no line.

**Comment by NiLuJe** [comment]
So, if you're comfortable with your line spacing and font sizes, what you can do is play with the top & bottom margins.

Specifically, lower the top margin a bit to give some room, and lower the bottom margin by *a lot* to allow for more room. If paragraph spacing enters the fray, it might even make sense to drop it to zero, to handle varying amounts of rivers.

**Comment by NiLuJe** [comment]
As for the first screenshot: widow & orphans, maybe?

Try the style tweak to ignore 'em.

**Comment by djtsorrell** [comment]
Thanks for all this. It's extremely helpful.

> You haven't shown us your margin settings, but assuming they're balanced, given your top margin and line-height: no, there isn't ;).

I said that top and bottom were 15 pt. If it helps, I have three grey boxes on my T/B margins. As for left and right, they are both 20 pt or four grey boxes.  I have now reduced the bottom margin by 5 pt.

As for the reclaim height setting, I now get why it was clipping. I didn't really understand what the setting was trying to do; I thought it just reduced white space. I have disabled it now and reduced my bottom margin and it's a lot better.
> (Or to murder that abomination with the dedicated style tweak ;p).

Oh yes, I was so happy to find that I could get rid of paragraph spacing. It was so ugly to have all these gaps.

> As for the first screenshot: widows & orphans, maybe?

Brilliant! This is exactly what I was looking for. I couldn't understand why it was cutting paragraphs and leaving all this white space but now I see it's part of this setting. Ignoring publisher settings has fixed my issue. Thank you!
